dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/entities/DataRequestEntity.kt

```diff
@@ -32,11 +39,64 @@ data class DataRequestEntity(
 
     val dataRequestCompanyIdentifierValue: String,
 
-    @Column(columnDefinition = "text")
-    var messageHistory: String?,
+    @OneToMany(mappedBy = "dataRequest")
+    var messageHistory: MutableList<MessageEntity>,
 
     val lastModifiedDate: Long,
 
     @Enumerated(EnumType.STRING)
     var requestStatus: RequestStatus,
-)
+) {
+
+    constructor(
+        userId: String,
+        dataType: DataTypeEnum,
+        reportingPeriod: String,
+        identifierType: DataRequestCompanyIdentifierType,
+        identifierValue: String,
+        currentTimestamp: Long = Instant.now().toEpochMilli(),
```

Comment:
Wäre es nicht besser, da keinen default Wert zu setzen?

Die Code-Stelle, welche die DataRequestEntity bauen möchte, sollte den timestamp selbst bestimmen.
Bei einer query durch Einlesen des DB-Eintrags, und beim storen durch Instant.now.

Ist auch bisschen Geschmacksache, aber es würde sich für mich sicherer anfühlen.

V.a. in solchen corne-cases wie "die Query findet keinen timestamp" => dann würde das Ding hier einfach die aktuelle Zeit reinschreiben, obwohl man dann vllt lieber einen Error hätte (?)

Was meinst du

Reply:
Mein Problem war, dass ich zwei Einträge im primary constructor habe, die diesen Wert haben müssen. Und ich kann bevor ich `this(...)` aufrufe keine variable anlegen. Daher habe ich das so gemacht. Soll ich lieber an zwei Stellen die epoch Methode aufrufen? Die minimale Differenz fällt wahrscheinlich nichtmal auf.

Reply:
Hab den Parameter entfernt.